### ISSUE TITLE
`-Split <Delimiter,<Max-substrings>` doesn't handle Non-Capturing Groups

### ISSUE THREAD

**Issue by iRon7** [PowerShell/PowerShell#24650]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```PowerShell
'abcd' -Split 'b(c)', 2
```

### Expected behavior

```console
a
c
```

### Actual behavior

```console
a
c
d
```

### Error details

```console
You might argue that it concerns sub-arrays but that is not visible in the end-result:

('abcd' -Split 'b(c)').Count
3
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.6
PSEdition                      Core
GitCommitId                    7.4.6
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
I think, this is because captured items are the part of result:

```
'abcd' -split '(bc)', 2
a
bc
d
```

You can escape them:
```
'abcd' -split '(?:bc)', 2
```

**Comment by iRon7** [comment]
@237dmitry,

Correct, the result of [non-capturing group](https://https//www.regular-expressions.info//brackets.html) is more what you might expect in return but that doesn't explane why a the indexing doesn't work. In fact, it lets me even wonder if there is any use of ***capturing* groups** using the `-Split` operator and whether the default shouldn't be ***non-capturing* group** (`ExplicitCapture`):

```PowerShell
'abcd' -Split 'b(c)', 2, 'ExplicitCapture'
```

Anyways, I still think it is a code smell.

**Comment by santisq** [comment]
The docs state:

> Specifies the maximum number of substrings returned by the split operation. The default is all substrings split by the delimiter. If there are more substrings, they are concatenated to the final substring.

I would have expected in this case:

```powershell
a
cd
```

**Comment by mklement0** [comment]
_De facto_ this is what the `-split` operator currently does (which may or may not be the correct behavior):

* It splits the input into as many tokens as requested by the `<Max-substrings>` option.

* _Independently_, it also returns whatever any _capturing_ groups in the splitting regex matched.

The following command demonstrates this:

```powershell
 '1-2-3-4' -split '(-)', 3
```

Despite asking for _3_ tokens, the resulting array comprises _5_ elements, because what `(-)` captured among the first _2_ matching separators (matching only _2_ separators is needed in order to produce _3_ tokens), is - perhaps unexpectedly - is included in the result array:

```none
1
-
2
-
3-4
```

**Comment by rkeithhill** [comment]
The Engine WG reviewed this and find that this is working as designed.  This behavior is also documented in:

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_split?view=powershell-7.5#max-substrings

and 

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_split?view=powershell-7.5#delimiter
